packages/trees/frontend/sirius-components-trees/src/trees/FilterBar.tsx

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import IconButton from '@material-ui/core/IconButton';
+import { Theme, makeStyles } from '@material-ui/core/styles';
+import CloseIcon from '@material-ui/icons/Close';
+import { FilterBarProps, FilterBarState } from './FilterBar.types';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import { FilterList } from '@material-ui/icons';
+import { useState } from 'react';
+import TextField from '@material-ui/core/TextField';
+
+const useFilterBarStyles = makeStyles((theme: Theme) => ({
+  filterbar: {
+    display: 'flex',
+    flexDirection: 'row',
+    overflow: 'hidden',
+    paddingLeft: theme.spacing(1),
+    paddingRight: theme.spacing(1),
+    paddingBottom: theme.spacing(1),
+  },
+  textfield: {
+    height: theme.spacing(3),
+    fontSize: theme.typography.fontSize,
+  },
+}));
+export const FilterBar = ({ onTextChange, onFilterButtonClick, onClose }: FilterBarProps) => {
+  const classes = useFilterBarStyles();
+
+  const initialState: FilterBarState = {
+    filterEnabled: false,
+  };
+  const [state, setState] = useState<FilterBarState>(initialState);
+
+  return (
+    <div className={classes.filterbar}>
+      <TextField
+        id={'filterbar-textfield'}
+        data-testid={'filterbar-textfield'}
+        name={'filterbar-textfield'}
+        placeholder={'Type to filter'}
```

Comment:
No need to perform some evaluation for such static piece of text, something like `id="..."` would be enough